src/main/java/com/rackspace/ceres/app/services/SuggestApiService.java

```diff
@@ -0,0 +1,75 @@
+package com.rackspace.ceres.app.services;
+
+import java.util.List;
+import javax.xml.crypto.Data;
+import jnr.ffi.annotations.Meta;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.cassandra.core.cql.ReactiveCqlTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Service
+public class SuggestApiService {
+
+  private final MetadataService metadataService;
+
+  @Autowired
+  public SuggestApiService(MetadataService metadataService) {
+    this.metadataService = metadataService;
+  }
+
+  /**
+   * Method used to return the list of tag keys associated with particular tenant with limit set to
+   * max
+   *
+   * @param tenant
+   * @param tagK
+   * @param max
+   * @return
+   */
+  public Mono<List<String>> suggestTagKeys(String tenant, String tagK, int max) {
+    return metadataService.getMetricNames(tenant).flatMapMany(Flux::fromIterable)
+        .flatMap(metric -> metadataService.getTagKeys(tenant, metric).flatMapMany(Flux::fromIterable)
+        .filter(tagKey -> !StringUtils.hasText(tagK) || tagKey.startsWith(tagK)))
```

Comment:
Why not do some regex here?

Reply:
Can you share an example of what do you mean by using regex here?

Reply:
```java
Pattern pattern = Pattern.compile("...");

List<String> matching = list.stream()
                            .filter(pattern.asPredicate())
                            .collect(Collectors.toList());
```

Reply:
Shouldn't we leave pattern matching with regex for the complex strings where java methods does not come handy?

Reply:
Sure, it was just a suggestion